### ISSUE TITLE
Cannot add 2 list_of_different_objects rules for the same field

### ISSUE THREAD

**Issue by mezzatto** [koorchik/js-validator-livr#29]
I have the following json:

```
{
  "id": 123,
  "pricingInfos": [
    {
      "period": "MONTHLY",
      "price": "20000",
      "businessType": "RENTAL",
    },
    {
      "price": "100000",
      "businessType": "SALE",
    }
  ]
}
```

I have this rule that adds 2 `list_of_different_objects` to the same `pricingInfos` field:

```
{
  "id": ["required", "positive_integer"],
  "pricingInfos": ["required",
    {
      "list_of_different_objects": ["period", {
          "DAILY": {"price": ["required", {"number_between": [3, 30000]}]},
          "WEEKLY": {"price": ["required", {"number_between": [25, 255000]}]},
          "MONTHLY": {"price": ["required", {"number_between": [100, 900000]}]},
          "YEARLY": {"price": ["required", {"number_between": [1200, 10800000]}]}
        }]
    },
    {
      "list_of_different_objects": ["businessType", {
          "SALE": {"price": ["required", {"number_between": [10000, 1000000000]}]},
          "RENTAL": {"period": "required"}
        }]
    }
  ]
}
```

Looks like it completely ignores both `list_of_different_objects` since the rules are not used (I can insert invalid values and no errors are detected).

Here is the playbook with that simulates the bug:

http://webbylab.github.io/livr-playground/#%7B%22rules%22%3A%22%7B%5Cn%20%20%5C%22id%5C%22%3A%20%5B%5C%22required%5C%22%2C%20%5C%22positive_integer%5C%22%5D%2C%5Cn%20%20%5C%22pricingInfos%5C%22%3A%20%5B%5C%22required%5C%22%2C%5Cn%20%20%20%20%7B%5Cn%20%20%20%20%20%20%5C%22list_of_different_objects%5C%22%3A%20%5B%5C%22period%5C%22%2C%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20%5C%22DAILY%5C%22%3A%20%7B%5C%22price%5C%22%3A%20%5B%5C%22required%5C%22%2C%20%7B%5C%22number_between%5C%22%3A%20%5B3%2C%2030000%5D%7D%5D%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20%5C%22WEEKLY%5C%22%3A%20%7B%5C%22price%5C%22%3A%20%5B%5C%22required%5C%22%2C%20%7B%5C%22number_between%5C%22%3A%20%5B25%2C%20255000%5D%7D%5D%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20%5C%22MONTHLY%5C%22%3A%20%7B%5C%22price%5C%22%3A%20%5B%5C%22required%5C%22%2C%20%7B%5C%22number_between%5C%22%3A%20%5B100%2C%20900000%5D%7D%5D%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20%5C%22YEARLY%5C%22%3A%20%7B%5C%22price%5C%22%3A%20%5B%5C%22required%5C%22%2C%20%7B%5C%22number_between%5C%22%3A%20%5B1200%2C%2010800000%5D%7D%5D%7D%5Cn%20%20%20%20%20%20%20%20%7D%5D%5Cn%20%20%20%20%7D%2C%5Cn%20%20%20%20%7B%5Cn%20%20%20%20%20%20%5C%22list_of_different_objects%5C%22%3A%20%5B%5C%22businessType%5C%22%2C%20%7B%5Cn%20%20%20%20%20%20%20%20%20%20%5C%22SALE%5C%22%3A%20%7B%5C%22price%5C%22%3A%20%5B%5C%22required%5C%22%2C%20%7B%5C%22number_between%5C%22%3A%20%5B10000%2C%201000000000%5D%7D%5D%7D%2C%5Cn%20%20%20%20%20%20%20%20%20%20%5C%22RENTAL%5C%22%3A%20%7B%5C%22period%5C%22%3A%20%5C%22required%5C%22%7D%5Cn%20%20%20%20%20%20%20%20%7D%5D%5Cn%20%20%20%20%7D%5Cn%20%20%5D%5Cn%7D%22%2C%22data%22%3A%22%7B%5Cn%20%20%5C%22id%5C%22%3A%20123%2C%5Cn%20%20%5C%22pricingInfos%5C%22%3A%20%5B%5Cn%20%20%20%20%7B%5Cn%20%20%20%20%20%20%5C%22period%5C%22%3A%20%5C%22MONTHLY%5C%22%2C%5Cn%20%20%20%20%20%20%5C%22price%5C%22%3A%20%5C%2220000%5C%22%2C%5Cn%20%20%20%20%20%20%5C%22businessType%5C%22%3A%20%5C%22RENTAL%5C%22%2C%5Cn%20%20%20%20%7D%2C%5Cn%20%20%20%20%7B%5Cn%20%20%20%20%20%20%5C%22price%5C%22%3A%20%5C%221%5C%22%2C%5Cn%20%20%20%20%20%20%5C%22businessType%5C%22%3A%20%5C%22SALE%5C%22%2C%5Cn%20%20%20%20%7D%5Cn%20%20%5D%5Cn%7D%22%7D

**Comment by mezzatto** [comment]
Got it! Thank you!
